packages/core/src/engines/Engine.ts

```diff
@@ -179,7 +179,7 @@ export abstract class Engine<Key extends GestureKey> {
    * Function ran at the start of the gesture.
    * @param event
    */
-  start(event: NonNullable<State[Key]>['event']) {
+  start(event: StateTypes[Key]['event']) {
```

Comment:
#### ref
- [TypeScript: TS Playground](https://www.typescriptlang.org/play?ts=4.8.0-dev.20220624&ssl=6&ssc=20&pln=6&pc=23#code/C4TwDgpgBAYg9nKBeKBvKBtA1hEAuKAZ2ACcBLAOwHMBdAgVwqwrgHcKoBfAWAChRIUAHL0ANqICGAI1ER4iFPKgAfKIwAmEAGaUI6vnwHQAQhJIAeACpQIAD2AQK6wsLGSZchAD5kwuBRFxaVkrLwxicmoaKAB6GIgSEjgSKH8oABYAOgAOABooFhtE5NSOLIB2A35wEwkALysbe0dnWG9fIX9A9xDLMIjKWliYgsQEpJJDGqgARXpbDq63YIhzbpX5ftJB6LjRoomgA)
-  `NonNullable<T>['event']` => Type 'string' cannot be used to index type 'NonNullable<T>'.(2536)